vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-gen/cmd/client-gen/generators/util/tags.go

```diff
@@ -146,16 +189,83 @@ func ParseClientGenTags(lines []string) (Tags, error) {
 		}
 		ret.SkipVerbs = skipVerbs
 	}
+	var err error
+	if ret.Expansions, err = parseClientExpansions(values); err != nil {
+		return ret, err
+	}
 	return ret, validateClientGenTags(values)
 }
 
+func parseClientExpansions(tags map[string][]string) ([]expansion, error) {
+	ret := []expansion{}
+	for name, values := range tags {
+		if !strings.HasPrefix(name, "genclient:") || !strings.HasSuffix(name, "Verb") {
+			continue
+		}
+		name = strings.TrimPrefix(name, "genclient:")
+		exp := expansion{}
+		for _, supportedVerb := range SupportedVerbs {
+			if strings.HasPrefix(name, supportedVerb) {
```

Comment:
this looks fuzzy. Can't we be more precise with matching?

Reply:
@sttts at this point we only matched everything that starts with the supported verb and has *Verb suffix. Everything between is considered as expansion verb name. Dunno if there is a better way other than regexp.